### PR TITLE
Tweaks, fixups, and general cleanup

### PR DESCRIPTION
**Pull request by blag**
**What this PR does / why we need it**:

* Fixes instructions for adding a new distro
* Documents all of the different types of builds and how they are triggered
* Always tags `main.dev` images
* Adds labels to images containing the CircleCI workflow ID that build the image and a link to the workflow
* Adjusts the suffix of the nightly build tags to `-nightly+YYYYMMDD`
* Uses the list of approvers that's already defined
* Removes (incorrect) tag from the Slack notification

### CODE REVIEW

File: .circleci/config.yml.j2 [hunk 1]
```diff
@@ -740,6 +754,9 @@ commands:
       edge_build:
         description: "Indicate if this is an edge build"
         type: boolean
+      nightly_build:
+        description: "Indicate if this is a nightly build"
+        type: boolean
```
Comment (kaxil) [inline .circleci/config.yml.j2:759]: We could probably add the default of `true` to be on the safer side, so we don't have to edit/update the non-nightlies

Comment (kaxil) [inline .circleci/config.yml.j2:759]:
```suggestion
        default: true
```

File: README.md [hunk 2]
```diff
@@ -4,6 +4,26 @@
...
+| `edge` build  | `main.dev`               | Built from the current `main` branch of [astronomer/airflow](https://github.com/astronomer/airflow) |
+| dev build     | `2.2.4-4.dev`            | Development build, released during ap-airflow changes, including pre-releases and version releases  |
+| nightly build | `2.2.4-nightly+20220314` | Nightly builds, regularly triggered by a CircleCI pipeline sometime during the midnight hour UTC    |
+| release build | `2.2.4-4`                | Release builds, triggered by a release PR                                                           |
+
+Note: Edge builds are always development builds
+
+#### Build matrix
+
+| Build         | Nightly                   | Pre-release PR     | Release PR         |
```
Comment (kaxil) [inline README.md:20]: This looks awesome, thanks for adding it

**Review by kaxil — APPROVED** [review]

File: .circleci/config.yml.j2 (hunk 1 above)
Comment (blag) [reply .circleci/config.yml.j2:759]: If I understand this correctly, I think setting the default to `true` would force us to update the non-nightlies. Setting it to `false` would let us omit that in the non-nightlies.

**Pull request merged**